fix: update hard-coded google model names

mfleader opened the pull request:
# What does this PR do?
When we send the model names to Google's openai API, we must use the
"google" name prefix. Google does not recognize the "vertexai" model
names.

Closes https://github.com/llamastack/llama-stack/issues/4211

## Test Plan
```bash
uv venv --python python312
. .venv/bin/activate
llama stack list-deps starter | xargs -L1 uv pip install
llama stack run starter
```

Test that this shows the gemini models with their correct names:
```bash
curl http://127.0.0.1:8321/v1/models | jq '.data | map(select(.custom_metadata.provider_id == "vertexai"))'
```

Test that this chat completion works:
```bash
curl -X POST   -H "Content-Type: application/json"   "http://127.0.0.1:8321/v1/chat/completions"   -d '{
        "model": "vertexai/google/gemini-2.5-flash",
        "messages": [
          {
            "role": "system",
            "content": "You are a helpful assistant."
          },
          {
            "role": "user",
            "content": "Hello! Can you tell me a joke?"
          }
        ],
        "temperature": 1.0,
        "max_tokens": 256
      }'
```

Also updates the pyproject version.

leseb reviewed (changes requested):
Thanks! I need to open a 2.24 branch and will change the target once done

ktdreyer committed:
fix: update hard-coded google model names (#4212)

When we send the model names to Google's openai API, we must use the "google" name prefix. Google does not recognize the "vertexai" model names.

Closes #4211

(cherry picked from commit dabebdd2303af1401c5dd9d92654b530c4db5050)

mfleader committed:
chore: Update pyproject.toml version to 0.3.0.1rc3+rhai0

Signed-off-by: Matthew F Leader <mleader@redhat.com>

leseb reviewed (commented):
You need to run pre-commit locally to update the uv.lock, also it looks pyproject has conflicts

leseb commented on pyproject.toml:
```diff
@@ -7,7 +7,7 @@ required-version = ">=0.7.0"
 [project]
 name = "llama_stack"
-version = "0.3.0rc3+rhai0"
+version = "0.3.0rc4+rhai0"
```
```suggestion
version = "0.3.0.1+rhai0"
```

actually the tag will use a fourth digit, we did that for the https://github.com/opendatahub-io/llama-stack/releases/tag/v0.2.22.2%2Brhai0
